internal/providers/terraform/aws/elasticsearch_domain.go

```diff
@@ -0,0 +1,161 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/pkg/schema"
+	"github.com/shopspring/decimal"
+)
+
+func NewElasticsearchDomain(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+
+	region := d.Get("region").String()
+	clusterConfig := d.Get("cluster_config").Array()[0]
+	instanceType := clusterConfig.Get("instance_type").String()
+	instanceCount := clusterConfig.Get("instance_count").Int()
+	dedicatedMasterEnabled := clusterConfig.Get("dedicated_master_enabled").Bool()
+	dedicatedMasterType := clusterConfig.Get("dedicated_master_type").String()
+	dedicatedMasterCount := clusterConfig.Get("dedicated_master_count").Int()
+	ultrawarmEnabled := clusterConfig.Get("warm_enabled").Bool()
+	ultrawarmType := clusterConfig.Get("warm_type").String()
+	ultrawarmCount := clusterConfig.Get("warm_count").Int()
+
+	ebsOptions := d.Get("ebs_options").Array()[0]
+
+	ebsTypeMap := map[string]string{
+		"gp2":      "GP2",
+		"io1":      "PIOPS-Storage",
+		"standard": "Magnetic",
+	}
+
+	gbVal := decimal.NewFromInt(int64(defaultVolumeSize))
+	if ebsOptions.Get("volume_size").Exists() {
+		gbVal = decimal.NewFromFloat(ebsOptions.Get("volume_size").Float())
+	}
+
+	ebsType := "gp2"
+	if ebsOptions.Get("volume_type").Exists() {
+		ebsType = ebsOptions.Get("volume_type").String()
+	}
+
+	ebsFilter := "gp2"
+	if val, ok := ebsTypeMap[ebsType]; ok {
+		ebsFilter = val
+	}
+
+	iopsVal := decimal.NewFromInt(1)
+	if ebsOptions.Get("iops").Exists() {
+		iopsVal = decimal.NewFromFloat(ebsOptions.Get("iops").Float())
+
+		if iopsVal.LessThan(decimal.NewFromInt(1)) {
+			iopsVal = decimal.NewFromInt(1)
+		}
+	}
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           fmt.Sprintf("Instance on-demand, (%s)", instanceType),
```

Comment:
I think `fmt.Sprintf("Instance (on-demand, %s)", instanceType),` would make it consistent with EC2/Mongo (same issue with the other two cases)

Reply:
Haha so close, sorry my bad, should have checked the other resources.

Reply:
Thanks, I'll merge as soon as that's done.